raiko-guest/src/one_shot.rs

```diff
@@ -27,21 +31,72 @@ use crate::{
 pub const ATTESTATION_QUOTE_DEVICE_FILE: &str = "/dev/attestation/quote";
 pub const ATTESTATION_TYPE_DEVICE_FILE: &str = "/dev/attestation/attestation_type";
 pub const ATTESTATION_USER_REPORT_DATA_DEVICE_FILE: &str = "/dev/attestation/user_report_data";
+pub const BOOTSTRAP_INFO_FILENAME: &str = "bootstrap.json";
 pub const PRIV_KEY_FILENAME: &str = "priv.key";
 
-pub fn bootstrap(global_opts: GlobalOpts) -> Result<()> {
-    let privkey_path = global_opts.secrets_dir.join(PRIV_KEY_FILENAME);
-    let key_pair = generate_key();
-    let mut file =
-        fs::File::create(&privkey_path).with_context(|| "Failed to create private key file")?;
+#[derive(Serialize)]
+struct BootstrapData {
+    public_key: String,
+    new_instance: Address,
+    #[serde(with = "Base64Standard")]
+    quote: Vec<u8>,
+}
+
+fn save_priv_key(key_pair: &KeyPair, privkey_path: &PathBuf) -> Result<()> {
+    let mut file = fs::File::create(privkey_path).with_context(|| {
+        format!(
+            "Failed to create private key file {}",
+            privkey_path.display()
+        )
+    })?;
     let permissions = std::fs::Permissions::from_mode(0o600);
     file.set_permissions(permissions)
-        .with_context(|| "Failed to set permissions to private key file")?;
+        .context("Failed to set restrictive permissions of the private key file")?;
     file.write_all(&key_pair.secret_bytes())
-        .with_context(|| format!("Failed to write to {}", privkey_path.display()))?;
+        .context("Failed to save encrypted private key file")?;
+    Ok(())
+}
+
+fn get_sgx_quote() -> Result<Vec<u8>> {
+    let mut quote_file = File::open(ATTESTATION_QUOTE_DEVICE_FILE)?;
+    let mut quote = Vec::new();
+    quote_file.read_to_end(&mut quote)?;
+    Ok(quote)
+}
+
+fn save_bootstrap_details(
+    key_pair: &KeyPair,
+    new_instance: Address,
+    quote: Vec<u8>,
+    bootstrap_details_file_path: &Path,
+) -> Result<(), Error> {
+    let bootstrap_details = BootstrapData {
+        public_key: key_pair.public_key().to_string(),
+        new_instance,
+        quote,
+    };
+    let json = serde_json::to_string_pretty(&bootstrap_details)?;
+    fs::write(bootstrap_details_file_path, json).context(format!(
+        "Saving bootstrap data file {} failed",
+        bootstrap_details_file_path.display()
+    ))?;
+    Ok(())
+}
+
+pub fn bootstrap(global_opts: GlobalOpts) -> Result<()> {
+    let key_pair = generate_key();
+    let privkey_path = global_opts.secrets_dir.join(PRIV_KEY_FILENAME);
+    save_priv_key(&key_pair, &privkey_path)?;
     println!("Public key: 0x{}", key_pair.public_key());
     let new_instance = public_key_to_address(&key_pair.public_key());
     println!("Instance address: {}", new_instance);
+    let quote = get_sgx_quote()?;
+    let bootstrap_details_file_path = global_opts.config_dir.join(BOOTSTRAP_INFO_FILENAME);
```

Comment:
It's not a good idea to store the privacy data into config path

Reply:
I partially agree: secrets data are typically save in `$HOME/.config` configuration directory instead of system-wide `/etc`. I fixed that - let me know if know is better. 👍🏼